components/settings/tabs/Account.tsx

```diff
@@ -0,0 +1,442 @@
+  IconCheck,
+  IconCheckbox,
+  IconFileUpload,
+  IconNotesOff,
+  IconX,
+} from '@tabler/icons-react';
+import { useSession } from 'next-auth/react';
+import React, { useRef, useState } from 'react';
+
+interface FormValues {
+  firstName?: string;
+  lastName?: string;
+  phone?: string;
+  school?: string;
+  bio?: string;
+  githubURL?: string;
+  linkedinURL?: string;
+  skills: string[];
+}
+
+export default function Account() {
+  // Component level states
+  const { data: session, status, update } = useSession();
+  const [isLoading, { open, close }] = useDisclosure();
+
+  // Avatar states, loading, and values
+  const inputFileRef = useRef<HTMLInputElement>(null);
+  const [loadingAvatar, setLoadingAvatar] = useState<boolean>(false);
+  const [isAvatarHovered, setIsAvatarHovered] = useState<boolean>(false);
+  const MAX_IMAGE_SIZE_MB = 4.5;
+  const MAX_FILE_SIZE_BYTES = 1000000 * MAX_IMAGE_SIZE_MB;
+
+  // Form states for editables
+  const [bioLength, setBioLength] = useState<number>(
+    session?.user?.bio?.length || 0
+  );
+  const [currentSkills, setCurrentSkills] = useState<string[]>(
+    session?.user?.skills || []
+  );
+  const [currentSkillError, setcurrentSkillError] = useState<boolean>(false);
+  const [formChanged, setFormChanged] = useState<boolean | null>(null);
+
+  const displayPhone = (phone: string) => {
+    if (!phone || phone.length !== 10) return null;
+
+    return `(${phone.slice(0, 3)})-${phone.slice(3, 6)}-${phone.slice(6)}`;
+  };
+
+  // Form initilization
+  const form = useForm({
+    mode: 'uncontrolled',
+    initialValues: {
+      bio: session?.user?.bio || '',
+      skills: currentSkills,
+    },
+    validate: yupResolver(profileConfigurationScheme),
+    onValuesChange: (values: any) => {
+      console.log(values);
+      if (values.bio !== null) setBioLength(values.bio.length);
+      setFormChanged(true);
+    },
+  });
+
+  // Preprocessing form values after submit
+  const cleanFormValues = (values: FormValues): Partial<FormValues> => {
+    const cleanedEntries = Object.entries(values).map(([key, value]) => {
+      if (key === 'phone' && typeof value === 'string') {
+        // Filter pure numbers out of phone number
+        const cleanedPhone = value.replace(/\D/g, '');
+        return [key, cleanedPhone];
+      }
+
+      if (Array.isArray(value)) {
+        // Leave untouched
+        return [key, value];
+      }
+
+      if (typeof value === 'string' && value.trim() === '') {
+        // Label empty strings as null for removal later
+        return null;
+      }
+
+      return [key, value];
+    });
+
+    // Filter out null entries (those that were empty strings)
+    return Object.fromEntries(cleanedEntries.filter((entry) => entry !== null));
+  };
+
+  // Submitting Form
+  const onSubmitForm = async (values: any) => {
+    open();
+
+    if (!session?.user?.id) {
+      console.error(
+        '[ERROR] No session ID. Please contact Technical Staff for help'
```

Comment:
People shouldn't be reading this, generally. It could just be "Cannot submit form without a session ID"